spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -407,8 +407,25 @@ class Analyzer(tableUtils: TableUtils,
         )
       }
     }
+    val aggMetadata: ListBuffer[AggregationMetadata] = if (joinConf.hasDerivations) {
+      val keyAndPartitionFields =
+        leftDf.schema.fields ++ Seq(org.apache.spark.sql.types.StructField(tableUtils.partitionColumn, StringType))
+      val sparkSchema = {
+        val schema: Seq[(String, DataType)] = leftSchema.toSeq ++ rightSchema.toSeq
+        StructType(SparkConversions.fromChrononSchema(schema))
+      }
+      val dummyOutputDf = tableUtils.sparkSession
+        .createDataFrame(tableUtils.sparkSession.sparkContext.parallelize(immutable.Seq[Row]()), sparkSchema)
+      val finalOutputColumns = joinConf.derivationsScala.finalOutputColumn(dummyOutputDf.columns).toSeq
+      val derivedDummyOutputDf = dummyOutputDf.select(finalOutputColumns: _*)
+      val columns = SparkConversions.toChrononSchema(
+        StructType(derivedDummyOutputDf.schema.filterNot(f => keyAndPartitionFields.map(_.name).contains(f.name))))
+      ListBuffer(columns.map { tup => toAggregationMetadata(tup._1, tup._2, joinConf.hasDerivations) }: _*)
+    } else {
+      aggregationsMetadata
```

Comment:
nit:  rename agg because agg is specific to group_by, but now we have external parts and derivations
- `aggMetadata` => `joinOutputsMetadata`
- `aggregationsMetadata` => `joinIntermediateFieldsMetadata`

Reply:
The concept of this aggMetadata is more like output features(not including keys). It will be used as the source data of features on MLI tool. I am thinking of maybe renaming this to featuresMetadata or joinOutputValuesMetadata.

Reply:
@yuli-han rename? 
- `aggMetadata` => `joinOutputValuesMetadata`
- `aggregationsMetadata` => `joinIntermediateValuesMetadata`